staging/src/k8s.io/apiserver/pkg/storage/value/encrypt/envelope/kmsv2/envelope.go

```diff
@@ -242,3 +250,22 @@ func generateKey(length int) (key []byte, err error) {
 
 	return key, nil
 }
+
+// validateAnnotations tests the following:
+//  1. The annotations key is what Kubernetes calls a "qualified name".
+//     This validation is the same as the label key name validation.
+//  2. The size of annotations keys + values is less than 32KB.
+func validateAnnotations(annotations map[string][]byte) error {
+	var errs []error
+	var totalSize int
+	for k, v := range annotations {
+		for _, msg := range validation.IsQualifiedName(k) {
```

Comment:
`IsFullyQualifiedDomainName` requires a `field.Path` as input and since we don't have an actual field here, what would you recommend we use?

Reply:
I would just base it off the proto API's struct tags:

https://github.com/kubernetes/kubernetes/blob/dfa4143086bf504c6c72d5eee8a2210b8ed41b9a/staging/src/k8s.io/apiserver/pkg/storage/value/encrypt/envelope/v2alpha1/api.pb.go#L284-L293

So I think in this case it ends up simply being `annotations` since that is a top level field from the response.